scylla-cql/src/frame/request/execute.rs

```diff
@@ -52,7 +54,77 @@ impl DeserializableRequest for Execute<'_> {
     }
 }
 
+/// CQL protocol-level representation of an `EXECUTE` request,
+/// used to execute a single prepared statement.
+#[cfg_attr(test, derive(Debug, PartialEq, Eq))]
+pub struct ExecuteV2<'a> {
+    /// ID of the prepared statement to execute.
+    pub id: cow_bytes::CowBytes<'a>,
+
+    /// ID of the result metadata stored locally.
+    pub result_metadata_id: Option<cow_bytes::CowBytes<'a>>,
+
+    /// Various parameters controlling the execution of the statement.
+    pub parameters: query::QueryParameters<'a>,
+}
+
+impl SerializableRequest for ExecuteV2<'_> {
+    const OPCODE: RequestOpcode = RequestOpcode::Execute;
+
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), CqlRequestSerializationError> {
+        // Serializing statement id
+        types::write_short_bytes(self.id.as_ref(), buf)
+            .map_err(ExecuteSerializationError::StatementIdSerialization)?;
+
+        // Serializing result metadata id
+        if let Some(id) = self.result_metadata_id.as_ref() {
+            types::write_short_bytes(id.as_ref(), buf)
+                .map_err(ExecuteSerializationError::ResultMetadataIdSerialization)?;
+        }
+
+        // Serializing params
+        self.parameters
+            .serialize(buf)
+            .map_err(ExecuteSerializationError::QueryParametersSerialization)?;
+        Ok(())
+    }
+}
+
+impl DeserializableRequest for ExecuteV2<'_> {
+    fn deserialize(buf: &mut &[u8]) -> Result<Self, RequestDeserializationError> {
+        let id = CowBytes::from(types::read_short_bytes(buf)?).into_owned();
+        let result_metadata_id = CowBytes::from(types::read_short_bytes(buf)?).into_owned();
+        let parameters = QueryParameters::deserialize(buf)?;
+
+        Ok(Self {
+            id,
+            result_metadata_id: Some(result_metadata_id),
+            parameters,
+        })
+    }
```

Comment:
:thinking: What about lifetimes here?
1. You're using an anonymous lifetime parameter. This AFAIU means that `deserialize` must satisfy the arbitrary anonymous lifetime specified in the `impl` block; in particular, `Self` must not borrow from `buf`. Do you agree?
2. If so, then it follows that `'_` must actually be `'static`. Then let's write `'_static` there.
3. Otherwise - if you wanted `deserialize` to allow borrowed deserialization (why not BTW?), then let's make the returned value borrow from `buf`.

Reply:
I did not put much effort into those impls because they are only used by proxy, so I just did the same thing that impl for `Execute` does.
I'll add explicit static. I dont see how to allow borrowed deserialization - if we tie output lifetime to input lifetime, then we need a static slice to get a static ExecuteV2.